src/__tests__/host-text-nesting.test.tsx

```diff
@@ -0,0 +1,185 @@
+      'pressable'
+    );
+  });
+
+  test('pressable within text, with multiple text children', () => {
+    const view = render(
+      <Text>
+        <Pressable testID="pressable" accessibilityRole="button">
+          <Text>Save</Text>
+          <Text>render</Text>
+        </Pressable>
+      </Text>
+    );
+
+    expect(view.getByRole('button', { name: 'Save' }).props.testID).toBe(
+      'pressable'
+    );
+  });
+
+  test('pressable within View, with text child', () => {
+    const view = render(
+      <View>
+        <Pressable testID="pressable" accessibilityRole="button">
+          <Text>Save</Text>
+        </Pressable>
+      </View>
+    );
+
+    expect(view.getByRole('button', { name: 'Save' }).props.testID).toBe(
+      'pressable'
+    );
+  });
+
+  test('pressable within View, with text child within view', () => {
+    const view = render(
+      <View>
+        <Pressable testID="pressable" accessibilityRole="button">
+          <View>
+            <Text>Save</Text>
+          </View>
+        </Pressable>
+      </View>
+    );
+
+    expect(view.getByRole('button', { name: 'Save' }).props.testID).toBe(
+      'pressable'
+    );
+  });
+
+  test('pressable within View within Text, with text child within view', () => {
+    const view = render(
+      <Text>
+        <View>
+          <Pressable testID="pressable" accessibilityRole="button">
+            <View>
+              <Text>Save</Text>
+            </View>
+          </Pressable>
+        </View>
+      </Text>
+    );
+
+    expect(view.getByRole('button', { name: 'Save' }).props.testID).toBe(
+      'pressable'
+    );
+  });
+
+  test('Text within pressable', () => {
+    const view = render(
+      <Pressable testID="pressable" accessibilityRole="button">
+        <Text testID="text">Save</Text>
+      </Pressable>
+    );
+
+    expect(view.getByText('Save').props.testID).toBe('text');
+  });
+
+  test('Text within view within pressable', () => {
+    const view = render(
+      <Pressable testID="pressable" accessibilityRole="button">
+        <View>
+          <Text testID="text">Save</Text>
+        </View>
+      </Pressable>
+    );
+
+    expect(view.getByText('Save').props.testID).toBe('text');
+  });
+
+  test('View with text child', () => {
+    const view = render(
+      <View testID="view">
+        <Text>Save</Text>
+      </View>
+    );
+
+    expect(view.getByTestId('view').props.testID).toBe('view');
+  });
+
+  test('Text within view', () => {
+    const view = render(
+      <View>
+        <Text testID="text">Save</Text>
+      </View>
+    );
+
+    expect(view.getByTestId('text').props.testID).toBe('text');
+  });
+
+  test('Text within view within text', () => {
+    const view = render(
+      <Text>
+        <View>
+          <Text testID="text">Save</Text>
+        </View>
+      </Text>
+    );
+
+    expect(view.getByTestId('text').props.testID).toBe('text');
```

Comment:
Otherwise the test seems to be testing `getByTestId` itself.

Reply:
Yep this is true, but as raised [here](https://github.com/callstack/react-native-testing-library/pull/1139#discussion_r982614865), most cases should be covered in the relevant queries test suites.